src/Cardano/Wallet/CoinSelection/Fee.hs

```diff
@@ -341,3 +350,62 @@ computeFee (CoinSelection inps outs chgs) =
     -- by definition, impossible; unless we messed up real hard.
     collapse [] _ =
         invariant "outputs are bigger than inputs" (undefined) (const False)
+
+-- | A linear equation on the transaction size. Represents the @\s -> a + b*s@
+-- function where @s@ is the transaction size in bytes, @a@ and @b@ are
+-- constant coefficients.
+data TxSizeLinear = TxSizeLinear Double Double
+    deriving (Eq, Show)
+
+cardanoPolicy :: TxSizeLinear
+cardanoPolicy = TxSizeLinear 155381 43.946
+
+-- | Estimate the fee for a transaction that has @ins@ inputs
+--   and @length outs@ outputs. The @outs@ lists holds the coin value
+--   of each output.
+--
+--   NOTE: The average size of @Attributes AddrAttributes@ and
+--         the transaction attributes @Attributes ()@ are both hard-coded
+estimateCardanoFee
+    :: TxSizeLinear
+    -> (Tx, [TxWitness])
+    -> Fee
+estimateCardanoFee (TxSizeLinear a b) txWithWitness@(Tx inps _, _) =
+    trace ("totalPayload: "<> show totalPayload
+          <> " length : " <> show (BL.length $ toLazyByteString $ encodeSignedTx txWithWitness)) $
+    Fee $ ceiling $ a + b*totalPayload
+    where
+        totalPayload :: Double
+        totalPayload = fromIntegral
+            $ boundAddrAttrSize + boundTxAttrSize + boundSignatureWitnessSize + payloadFromTxWithWitness + 4*(length inps - 1)
+
+        payloadFromTxWithWitness :: Int
+        payloadFromTxWithWitness = (fromIntegral . BL.length . toLazyByteString . encodeSignedTx) txWithWitness
+
+        -- | Size to use for a value of type @Attributes AddrAttributes@ when estimating
+        --   encoded transaction sizes. The minimum possible value is 2.
+        --
+        -- NOTE: When the /actual/ size exceeds this bounds, we may underestimate
+        -- tranasction fees and potentially generate invalid transactions.
+        --
+        -- `boundAddrAttrSize` needed to increase due to the inclusion of
+        -- `NetworkMagic` in `AddrAttributes`. The `Bi` instance of
+        -- `AddrAttributes` serializes `NetworkTestnet` as [(Word8,Int32)] and
+        -- `NetworkMainOrStage` as []; this should require a 5 Byte increase in
+        --`boundAddrAttrSize`. Because encoding in unit tests is not guaranteed
+        -- to be efficient, it was decided to increase by 7 Bytes to mitigate
+        -- against potential random test failures in the future.
```

Comment:
I'd rather see a more concise comment explaining where the sizes come from:

```hs
-- - 34 => ?
-- - 7 => 5 (network magic) + 2 ?
```

Reply:
made more concise comments everywhere